### PR TITLE
[node-local-dns] fix incorrect 'pay attention' section

### PR DESCRIPTION
**Pull request by hugik-88**
## Description
This pull request corrects inaccuracies and lack of unambiguity of interpretation in the documentation of the node-local-dns module

## Why do we need it, and what problem does it solve?
Currently, we have some functionality that was not previously described in detail in the documentation. For example, the simultaneous operation of `cni-cilium` and `node-local-dns`. This real pull request provides more clarity on the specifics of this collaboration. Additionally, restrictions for certain CNIs, including `cni-simple-bridge` and `cni-flannel`, have been clarified.

## What is the expected result?
Currently, we have some functionality that was not previously described in detail in the documentation. For example, the simultaneous operation of `cni-cilium` and `node-local-dns`. This real pull request provides more clarity on the specifics of this collaboration. Additionally, restrictions for certain CNIs, including `cni-simple-bridge` and `cni-flannel`, have been clarified.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-local-dns
type: chore
summary: Documentation updates.
impact_level: low
```

### CODE REVIEW

File: ee/be/modules/350-node-local-dns/docs/CONFIGURATION_RU.md [hunk 1]
```diff
@@ -7,5 +7,5 @@ title: "Модуль node-local-dns: настройки"
 Модуль не требует конфигурации (все работает из коробки).
 
 **Обратите внимание на следующее:**
-- Модуль работает только для iptables-режима `kube-proxy` (ipvs не поддерживается и поведение с ipvs не проверялось).
-- По умолчанию модуль **не работает** для запросов из `hostNetwork`, все запросы уходят в `kube-dns`. В данном случае можно самостоятельно в конфигурации пода указать адрес `169.254.20.10`, но тогда в случае падения `node-local-dns` не будет работать fallback на `kube-dns`.
+- Модуль работает со всеми `CNI`, но для корректной работы с `cni-cilium` необходимо выполнить ряд [условий](../../../../../modules/021-cni-cilium/#ограничения).
```
Comment (z9r5) [inline ee/be/modules/350-node-local-dns/docs/CONFIGURATION_RU.md:10]:
```suggestion
- Модуль работает со всеми `CNI`, но для корректной работы с `cni-cilium` необходимо выполнить ряд [условий](../021-cni-cilium/#ограничения).
```

File: ee/be/modules/350-node-local-dns/docs/CONFIGURATION.md [hunk 2]
```diff
@@ -7,5 +7,5 @@ title: "The node-local-dns module: configuration"
 The module does not require any configuration (it works right out-of-the-box).
 
 **Pay attention to the following:**
-- The module supports the iptables `kube-proxy` mode only (the ipvs mode is not supported and not tested).
-- By default, the module **does not** serve `hostNetwork` requests (they are forwarded to `kube-dns`). In this case, you can specify the  `169.254.20.10`address in the Pod configuration yourself. However, if a `node-local-dns` will crash, you will not be able to get back to `kube-dns`.
+- The module works with all `CNIs`, but in order to work correctly with `cni-cilium`, a number of [conditions](../../../../../modules/021-cni-cilium/#limitations) must be met.
```
Comment (z9r5) [inline ee/be/modules/350-node-local-dns/docs/CONFIGURATION.md:10]:
```suggestion
- The module works with all `CNIs`, but in order to work correctly with `cni-cilium`, a number of [conditions](../021-cni-cilium/#limitations) must be met.
```

File: ee/be/modules/350-node-local-dns/docs/CONFIGURATION_RU.md [hunk 3]
```diff
@@ -7,5 +7,5 @@ title: "Модуль node-local-dns: настройки"
...
+- По умолчанию для `cni-simple-bridge` и `cni-flannel` модуль **не работает** для запросов из `hostNetwork`, все запросы уходят в `kube-dns`. В данном случае можно самостоятельно в конфигурации пода указать адрес `169.254.20.10`, но тогда в случае падения `node-local-dns` не будет работать `fallback` на `kube-dns`.
```
Comment (z9r5) [inline ee/be/modules/350-node-local-dns/docs/CONFIGURATION_RU.md:11]:
```suggestion
- По умолчанию, при использовании совместно с модулем `cni-simple-bridge` или `cni-flannel`, модуль `node-local-dns` **не работает** для запросов из `hostNetwork`. В этом случае все запросы уходят в модуль `kube-dns`. Вы можете  указать адрес `169.254.20.10` в конфигурации пода, но тогда, в случае проблем в работе модуля `node-local-dns`, не будет работать *fallback* на модуль `kube-dns`.
```

File: ee/be/modules/350-node-local-dns/docs/CONFIGURATION.md [hunk 4]
```diff
@@ -7,5 +7,5 @@ title: "The node-local-dns module: configuration"
...
+- By default for `cni-simple-bridge` and `cni-flannel`, the module **does not** serve `hostNetwork` requests (they are forwarded to `kube-dns`). In this case, you can specify the  `169.254.20.10` address in the Pod configuration yourself. However, if a `node-local-dns` will crash, you will not be able to get back to `kube-dns`.
```
Comment (z9r5) [inline ee/be/modules/350-node-local-dns/docs/CONFIGURATION.md:11]:
```suggestion
- By default, when used together with the `cni-simple-bridge` or `cni-flannel` modules, the `node-local-dns` module **does not work** for requests from `hostNetwork`. In this case, all requests go to the `kube-dns` module. You can specify the address `169.254.20.10` in the pod configuration, but if `node-local-dns` module crashes, there will be no *fallback* to the `kube-dns` module.
```

**Review by z9r5 — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [9504](https://github.com/deckhouse/deckhouse/pull/9504) to the branch [release-1.63](https://github.com/deckhouse/deckhouse/tree/release-1.63) successful!

**Pull request merged**